### PR TITLE
PBS-642 Fix Core Data Errors during unit tests

### PR DESCRIPTION
**Pull request by jbethanco**
Core Data does not cache the data model. During unit tests we bring up
 a second data controller for testing. This causes Core Data to load the
data model twice. It then sees conflicts between the two (same)
data models. By making the model a static let off of the dataController
class it can not be loaded twice.

Minor unrelated change:
Wrap  checkForUITestArguments() with Debug flag so it doesn't go into
production code.

### CODE REVIEW

**Comment by Pete-Hoch** [comment]
This looks fine, but I do not see the bug. How do I get the code to fail? All of our unit tests run fine with or without this patch.

File: PBLogging/App/SceneDelegate.swift [hunk 1]
```diff
@@ -21,7 +21,9 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
 
+        #if DEBUG
         checkForUITestArguments()
+        #endif
```
Comment (Pete-Hoch) [inline PBLogging/App/SceneDelegate.swift:26]: Good idea.

**Review by Pete-Hoch — APPROVED** [review]
Looks good.

**Review by petermisik — APPROVED** [review]
Looks good.

**Pull request merged**